driver/ixsystems/common.py

```diff
@@ -435,6 +438,48 @@ def _create_volume_from_snapshot(self, name, snapshot_name,
         except Exception as e:
             raise FreeNASApiError('Unexpected error', e)
 
+    def _promote_volume(self, volume_name):
+        """Promote a volume"""
+        request_urn = ('%s/id/%s/promote') % (
+            FreeNASServer.REST_API_VOLUME,
+            urllib.parse.quote_plus(
+                self.configuration.ixsystems_dataset_path + '/' + volume_name))
+        LOG.debug('_promote_volume urn : %s', request_urn)
+        try:
+            ret = self.handle.invoke_command(FreeNASServer.CREATE_COMMAND,
+                                             request_urn, None)
+            if ret['status'] != FreeNASServer.STATUS_OK:
+                msg = ('Error while promoting volume: %s' % ret['response'])
+                raise FreeNASApiError('Unexpected error', msg)
+        except Exception as e:
+            raise FreeNASApiError('Unexpected error', e)
+
```

Comment:
Use TrueNAS promote API to promote a volume.
Currently the only way to make this call is by upstream service account on a cloned cached volume.